src/main/java/net/fabricmc/loader/minecraft/McVersionLookup.java

```diff
@@ -65,6 +68,35 @@
 	private static final Pattern INDEV_PATTERN = Pattern.compile("(?:inf-|Inf?dev )(?:0\\.31 )?(\\d+(-\\d+)?)");
 	private static final String STRING_DESC = "Ljava/lang/String;";
 
+	public static OptionalInt getClassVersion(Path gameJar, List<String> entrypointClasses) {
+		try (FileSystemUtil.FileSystemDelegate jarFs = FileSystemUtil.getJarFileSystem(gameJar, false)) {
+			FileSystem fs = jarFs.get();
+
+			for (String entrypointClass : entrypointClasses) {
+				String fileString = entrypointClass.replace('.', '/') + ".class";
+				Path file = fs.getPath(fileString);
+
+				if (Files.isRegularFile(file)) {
+					DataInputStream is = new DataInputStream(Files.newInputStream(file));
+
+					if (is.readInt() != 0xCAFEBABE) {
+						continue;
```

Comment:
This wont close the DataInputStream, use try with resources?

Reply:
Fixed in 2b725fc9ead43dc3aaccc10b8c64287ff615f9b6